src/transformers/utils/generic.py

```diff
@@ -123,6 +123,24 @@ def is_torch_device(x):
     return False if not is_torch_available() else _is_torch_device(x)
 
 
+def _is_torch_dtype(x):
+    import torch
+
+    if isinstance(x, str):
+        if hasattr(torch, x):
+            return True
```

Comment:
I would be more careful -> i.e. to get that attribute, than test if it is indeed a `torch.dtype`

Reply:
you mean 
```
if hasattr(torch, x):
    x = getattr(torch, x)
else:
    return False
```
?

Reply:
yeah, `x = getattr(torch, x)` and then test if it is a torch dtype, just as what you did a few line below.

Be careful, I always complicate things. You can hold on until sylvain's review.

Reply:
Adapted your suggestion in [da35051](https://github.com/huggingface/transformers/pull/20536/commits/da35051ed08070eeff691891b5eb7232a1c12dc2) ;)